src/averageAges.js

```diff
@@ -55,7 +61,15 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const withSon = people.filter(person =>
+    people.find(mother => mother.name === person.mother) && person.sex === 'm');
+  const noSon = people.filter(person =>
+    people.find(mother => mother.name === person.mother));
+  const children = onlyWithSon ? withSon : noSon;
+  const age = children.map(child => (child.born - people.find(mother =>
+    mother.name === child.mother).born));
+
+  return age.reduce((a, b) => a + b) / age.length;
```

Comment:
you can create a function for repeating code, like this one `age.reduce((a, b) => a + b) / age.length`